### DISCUSSION TITLE
Docker container for whisper-gui

### DISCUSSION THREAD

**Discussion by 3x3cut0r** [Pikurrot/whisper-gui#25] [General]
Hey!
First of all i want to thank you for this great project.
I have created a [docker container](https://hub.docker.com/r/3x3cut0r/whisper-gui) of it.
It's a CPU only version of this project.
What do you think?

**Comment by Pikurrot** [comment]
Hi, it's great! I've tested it myself and works fine and smoothly.
However, I'm not very experienced on docker, and I'm having trouble trying to use a custom model I have saved locally. Have you been successful on that? How would you do it?

**Reply by 3x3cut0r** [reply]
No, i honestly didn't try it. But i will have a look on that in the next view days.

**Reply by Pikurrot** [reply]
Ok my bad, I had some weird errors but after updating docker desktop and restart the container it worked.
If you don't mind, I will link your dockerhub project in the Readme.
Now I was wondering, could we allow the container to use the host GPU to run whisper?

**Reply by 3x3cut0r** [reply]
the problem is i have only a mac mini and no cuda compatible gpu to test with. this problem i have also in other projects like privategpt ... and cuda needs another base image to run.
what i can do or test is to move the container to a cuda base image and after that i can test if its still working.

**Reply by Pikurrot** [reply]
ok try that. I myself have a computer with nvidia gpu, so I will try in the next days to allow gpu for the container.
My aim with this project is that it's compatible for any device, so this is important.

**Reply by 3x3cut0r** [reply]
So i want to build an image with the nvidia:cuda as base. Unfortunately my M2 Mac wont build it for some rosetta error: `rosetta error: failed to open elf at /lib64/ld-linux-x86-64.so.2`
Seems like the image is not compatible with MacOS. So i cannot continue with building an cuda image :/

Do you know maybe where i can find a free platform with nvidia gpus to test with? I tried google colab, but there it is not possible to install vanilla docker ...

**Reply by Pikurrot** [reply]
AWS has [several free tier offers](https://aws.amazon.com/free/), maybe some of them allow you a VM with gpu access, take a look but I don't know I've never used AWS.

I myself have not been able to successfully use gpu in a docker container. I can build the image but then when running the container I get some errors, mainly this one I've not been able to fix: `RuntimeError: CUDA failed with error CUDA driver version is insufficient for CUDA runtime version`.
Here is the dockerfile I'm using: https://github.com/Pikurrot/docker/blob/main/whisper-gui/Dockerfile

**Reply by KaKi87** [reply]
Hi,

I'm experiencing this very error you quoted using `WhisperGUI-cuda-x86_64.AppImage` from release v2.4.0 on Pop OS 24.04 with an AMD Ryzen 7 3700X CPU and an AMD Radeon RX 6700 XT GPU.

What to do ?

Thanks

**Reply by Pikurrot** [reply]
Hi, the AppImage build is only available for CUDA, which only works on Nvidia GPUs. Since you have a AMD GPU, this won't work on your machine. Try the alternative installation methods. I recommend you the wizard, which will identify your device as a AMD and should work correctly.
Tell me if you need further help :)
